pandas/core/groupby/groupby.py

```diff
@@ -690,7 +690,8 @@ def _set_result_index_ordered(self, result):
             result.set_axis(index, axis=self.axis, inplace=True)
             result = result.sort_index(axis=self.axis)
 
-        result.set_axis(self.obj._get_axis(self.axis), axis=self.axis, inplace=True)
+        obj = self.obj.dropna() if self.dropna else self.obj
```

Comment:
Another location where we don't want to just drop the null data?

Reply:
Yeah...

although we do have to do something about this line b/c `result` is an object with NA dropped and `self.obj` still has NA when this runs

Reply:
reverted this in 5074510, the changes we need here can be implemented in `_transform_general`